src/Microsoft.Framework.PackageManager/Utils/FrameworkSelectionHelper.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Versioning;
+using Microsoft.Framework.Runtime.Helpers;
+
+namespace Microsoft.Framework.PackageManager.Utils
+{
+    public static class FrameworkSelectionHelper
+    {
+        public static IEnumerable<FrameworkName> SelectFrameworks(Runtime.Project project,
```

Comment:
Well, then in this case consumer needs to create a list to store error messages and also need to find a proper way to display the list of messages, which defeats the purpose of simplify consumer codes. 

Both pattern works, they're generally the equivalent. Just one is already adopted and I don't see a strong case to change to another pattern.